File: leetcode/0287. Find the Duplicate Number/Solution.swift
// https://leetcode.com/problems/find-the-duplicate-number/

class Solution {
    func findDuplicate(_ nums: [Int]) -> Int {
        let nums = nums.sorted()
        for i in 1..<nums.count {
            if nums[i] == nums[i-1] {
                return nums[i]
            }
        }

        return -1
    }
}

File: leetcode/0287. Find the Duplicate Number/Solution.py
# https://leetcode.com/problems/find-the-duplicate-number/

from typing import List

class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        nums = sorted(nums)
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                return nums[i]

        return -1

s = Solution()
assert s.findDuplicate([1,3,4,2,2]) == 2
assert s.findDuplicate([3,1,3,4,2]) == 3